test/timezone.lisp

```diff
@@ -69,6 +69,20 @@
          (new (timestamp+ old 24 :hour eastern-tz)))
     (is (= (* 24 60 60) (timestamp-difference new old)))))
 
+(deftest test/timezone/parse-timestring-different-timezones ()
+   (let ((*default-timezone* eastern-tz))
+     (is (timestamp=
+          (parse-timestring "2016-12-01" :offset nil)                               ;; Use default timezone
+          (parse-timestring "2016-12-01" :offset (* 60 60 -5))                      ;; Ignore timezone 
+          (parse-timestring "2016-12-01" :offset (* 60 60 -5) :timezone utc-leaps)  ;; Ignore timezone
+          (parse-timestring "2016-12-01" :offset nil :timezone eastern-tz)          ;; Use timezone
```

Comment:
Wow, I've let this PR seriously languish. Sorry.

I mean that by setting *default-timezone* to eastern-tz and carefully arranging the arguments so that they return the same timestamps, this test basically doesn't show that the arguments do anything at all. A version of parse-timestring that ignored all its arguments and only used *default-timezone* would pass this test.